src/test/java/org/openrewrite/java/testing/cleanup/AssertionsArgumentOrderTest.java

```diff
@@ -54,7 +55,45 @@ String result() {
               """,
             """
               import static org.junit.jupiter.api.Assertions.assertEquals;
-                          
+
+              class MyTest {
+                  void someMethod() {
+                      assertEquals("result", result());
+                      assertEquals("result", result(), "message");
+                      assertEquals(0L, 1L);
+                  }
+                  String result() {
+                      return "result";
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite-testing-frameworks/issues/636")
+    void junit4AssertEqualsHavingPrimitiveArg() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              import static org.junit.Assert.assertEquals;
+
+              class MyTest {
+                  void someMethod() {
+                      assertEquals(result(), "result");
+                      assertEquals(result(), "result", "message");
```

Comment:
This line is converted into `assertEquals("result", result(), "message");`,
whereas argument order is: `assertEquals(message, expected, actual)`.

In this case we're making an incorrect change for JUnit 4 because the message used to be the first argument there, as opposed to the last argument with JUnit Jupiter. That's especially problematic for assertions on Strings, as the compiler does not warn you.